sherpa/astro/ui/tests/test_astro_ui_plot.py

```diff
@@ -2954,3 +2956,609 @@ def test_pha1_plot_multiple_args(clean_astro_ui, basic_pha1):
+    pdata = getattr(s, 'get_{}_plot'.format(plotfunc))
+
+    # Create the plot
+    plot()
+    p1 = pdata()
+    if plotfunc == 'fit':
+        if is_int:
+            assert p1.dataplot.histo_prefs['xlog']
+            assert p1.modelplot.histo_prefs['xlog']
+        else:
+            assert p1.dataplot.plot_prefs['xlog']
+            assert p1.modelplot.plot_prefs['xlog']
+    elif plotfunc in ['resid', 'ratio', 'delchi']:
+        assert p1.plot_prefs['xlog']
+    elif is_int:
+        assert p1.histo_prefs['xlog']
+    else:
+        assert p1.plot_prefs['xlog']
+
+    s.set_xlinear()
+    plot()
+
+    # Technically not needed as p1 is the same as p2
+    p2 = pdata()
+    if plotfunc == 'fit':
+        if is_int:
+            assert not p2.dataplot.histo_prefs['xlog']
+            assert not p2.modelplot.histo_prefs['xlog']
+        else:
+            assert not p2.dataplot.plot_prefs['xlog']
+            assert not p2.modelplot.plot_prefs['xlog']
+    elif plotfunc in ['resid', 'ratio', 'delchi']:
+        assert not p2.plot_prefs['xlog']
+    elif is_int:
+        assert not p2.histo_prefs['xlog']
+    else:
+        assert not p2.plot_prefs['xlog']
+
+
+@pytest.mark.parametrize("cls",
+                         [sherpa.ui.utils.Session, sherpa.astro.ui.utils.Session])
+@pytest.mark.parametrize("datafunc", [example_data1d,
+                                      example_data1dint])
+@pytest.mark.parametrize("plotfunc,answer",
+                         [('data', True),
+                          ('model', True),
+                          ('source', True),
+                          ('fit', True),
+                          ('resid', False),
+                          ('ratio', False),
+                          ('delchi', False)
+                         ])
+def test_set_plot_opt_y(cls, datafunc, plotfunc, answer):
+    """Does set_ylog/ylinear work?
+
+    Unlike test_set_plot_opt_x, the Y axis setting of the plot
+    does not necessarily folloe the set_ylog setting (e.g.
+    the residual plots).
+    """
+
+    s = cls()
+    s._add_model_types(basic)
+
+    s.set_ylog()
+
+    # load the data *after* calling the log method, as it should
+    # not matter what the data type is
+    #
+    s.set_data(datafunc())
+    is_int = hasattr(s.get_data(), 'xlo')
+
+    # Set up a model, in case we need it.
+    mdl = s.create_model_component('polynom1d', 'm1')
+    mdl.c0 = 1
+    mdl.c1 = 1
+    s.set_source(mdl)
+
+    plot = getattr(s, 'plot_{}'.format(plotfunc))
+    pdata = getattr(s, 'get_{}_plot'.format(plotfunc))
+
+    plot()
+    p1 = pdata()
+    if plotfunc == 'fit':
+        if is_int:
+            assert p1.dataplot.histo_prefs['ylog'] == answer
+            assert p1.modelplot.histo_prefs['ylog'] == answer
+        else:
+            assert p1.dataplot.plot_prefs['ylog'] == answer
+            assert p1.modelplot.plot_prefs['ylog'] == answer
+    elif plotfunc in ['resid', 'ratio', 'delchi']:
```

Comment:
Will the assertions ever change here? If not, would it be more efficient to remove the 
 
    elif plotfunc in ['resid', 'ratio', 'delchi']:
        assert p1.plot_prefs['ylog'] == answer

since the same code is executed in the default/fallthrough case.

Reply:
It actually isn't the same as the fall through case. since this check needs to be run before the is_int check (ie if you take it out the test fails). I've added a comment to note this.

Reply:
I perhaps should have added it to other tests that use the same ordering, but there's unfortunately a lot of similar-but-not-quite-similar-enough code that I felt it worth abstracting over (since I've found that just gets really hard to follow).